app/models/dog.rb

```diff
@@ -1,2 +1,8 @@
 class Dog < ApplicationRecord
+
+	validates :name, presence: true
+	validates :gender, presence: true
```

Comment:
Instead of simply requiring a value here, why not require one of two specific values -- ('male' or 'female') -- to help keep the data in the database clean?

See http://guides.rubyonrails.org/active_record_validations.html#inclusion for an example